src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +241,503 @@ otError DuaManager::Store(void)
+        SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kMeshLocalEid, &child->GetMeshLocalIid(),
+                                             sizeof(Ip6::InterfaceIdentifier)));
+
+        lastTransactionTime = Time::MsecToSec(TimerMilli::GetNow() - child->GetLastHeard());
+        SuccessOrExit(error = Tlv::AppendUint32Tlv(*message, ThreadTlv::kLastTransactionTime, lastTransactionTime));
+#endif // OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    }
+
+    if (!mle.IsFullThreadDevice() && mle.GetParent().IsThreadVersion1p1())
+    {
+        uint8_t pbbrServiceId;
+
+        SuccessOrExit(error = Get<BackboneRouter::Leader>().GetServiceId(pbbrServiceId));
+        SuccessOrExit(error = mle.GetServiceAloc(pbbrServiceId, messageInfo.GetPeerAddr()));
+    }
+    else
+    {
+        messageInfo.GetPeerAddr().SetToRoutingLocator(mle.GetMeshLocalPrefix(),
+                                                      Get<BackboneRouter::Leader>().GetServer16());
+    }
+
+    messageInfo.SetPeerPort(kCoapUdpPort);
+    messageInfo.SetSockAddr(Get<Mle::MleRouter>().GetMeshLocal16());
+
+    SuccessOrExit(error = Get<Coap::Coap>().SendMessage(*message, messageInfo, &DuaManager::HandleDuaResponse, this));
+
+    mIsDuaPending = true;
+
+    // TODO: (DUA) need update when CSL is enabled.
+    if (!Get<Mle::Mle>().IsRxOnWhenIdle())
+    {
+        Get<DataPollSender>().SendFastPolls();
+    }
+
+exit:
+    if (error == OT_ERROR_NO_BUFS)
+    {
+        UpdateCheckDelay(Mle::kNoBufDelay);
+    }
+
+    if (error != OT_ERROR_NONE && message != nullptr)
+    {
+        message->Free();
+    }
+
+    otLogInfoDua("Sent DUA.req for DUA %s: %s", dua.ToString().AsCString(), otThreadErrorToString(error));
+}
+
+void DuaManager::HandleDuaResponse(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo, otError aResult)
+{
+    OT_UNUSED_VARIABLE(aMessageInfo);
+    otError error;
+
+    mIsDuaPending = false;
+
+    if (aResult == OT_ERROR_RESPONSE_TIMEOUT)
+    {
+        UpdateCheckDelay(Mle::KResponseTimeoutDelay);
+        ExitNow(error = aResult);
+    }
+
+    VerifyOrExit(aResult == OT_ERROR_NONE && aMessage.GetCode() == OT_COAP_CODE_CHANGED, error = OT_ERROR_PARSE);
+
+    error = ProcessDuaResponse(aMessage);
+
+exit:
+    if (error != OT_ERROR_RESPONSE_TIMEOUT)
+    {
+        mRegistrationTask.Post();
+    }
+
+    otLogInfoDua("Received DUA.req: %s", otThreadErrorToString(error));
+}
+
+void DuaManager::HandleDuaNotification(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    OT_UNUSED_VARIABLE(aMessageInfo);
+    otError error;
+
+    OT_UNUSED_VARIABLE(error);
+
+    VerifyOrExit(aMessage.GetCode() == OT_COAP_CODE_POST, error = OT_ERROR_PARSE);
+
+    if (aMessage.IsConfirmable())
+    {
+        if (Get<Coap::Coap>().SendEmptyAck(aMessage, aMessageInfo) == OT_ERROR_NONE)
+        {
+            otLogInfoDua("Sent DUA.ntf acknowledgment");
+        }
+    }
```

Comment:
Combining the `if()`?
```suggestion
    if (aMessage.IsConfirmable() && (Get<Coap::Coap>().SendEmptyAck(aMessage, aMessageInfo) == OT_ERROR_NONE))
    {
        otLogInfoDua("Sent DUA.ntf acknowledgment");
    }
```